### PR TITLE
Design language groundwork

### PR DESCRIPTION
**Pull request by boyoonc**

### CODE REVIEW

**Comment by boyoonc** [comment]
I have a question on the accompanying mmui-web changes, before I issue a pull request for it!

File: gulpfile.js [hunk 1]
```diff
@@ -41,20 +42,31 @@ gulp.task('sass-app1', function () {
 
 });
 
+gulp.task('sass-design-language', function(){
```
Comment (besseym) [inline gulpfile.js]: we could probably just call it signal now

File: static/design_language/sass/design_language.scss [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+@import "../../../node_modules/bootstrap/scss/close";
+@import "../../../node_modules/bootstrap/scss/modal";
+@import "../../../node_modules/bootstrap/scss/tooltip";
+@import "../../../node_modules/bootstrap/scss/popover";
+@import "../../../node_modules/bootstrap/scss/carousel";
+@import "../../../node_modules/bootstrap/scss/utilities";
+
+.header{
+  //bootstrap
+  @extend .nav-link; //I hope I won't regret this decision
```
Comment (besseym) [inline static/design_language/sass/design_language.scss]: Regret this decision? It does look strange that the header is extending nav-link.

File: static/design_language/sass/design_language.scss [hunk 3]
```diff
@@ -0,0 +1,191 @@
+
+@import "../../../node_modules/bootstrap/scss/functions";
```
Comment (besseym) [inline static/design_language/sass/design_language.scss]: Can this entire first block be put into a sass file in mmui?

File: static/design_language/sass/design_language.scss [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+@import "../../../node_modules/bootstrap/scss/tooltip";
+@import "../../../node_modules/bootstrap/scss/popover";
+@import "../../../node_modules/bootstrap/scss/carousel";
+@import "../../../node_modules/bootstrap/scss/utilities";
+
+.header{
+  //bootstrap
+  @extend .nav-link; //I hope I won't regret this decision
+  //spacing
+  line-height: 22px; //To give same height as sub-navs
```
Comment (besseym) [inline static/design_language/sass/design_language.scss]: I like the extensive use of comments.

File: static/design_language/sass/design_language.scss (hunk 3 above)
Comment (besseym) [reply static/design_language/sass/design_language.scss]: Maybe we have talked about this before.

File: static/design_language/sass/design_language.scss [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+  padding-left: 0;
+  padding-bottom: 18px; //40px(design for between subheaders)-22px(body-copy line-height, as it is the font of the following line)
+  padding-top: 0px;
+  //icon
+  i{
+    color: $mm-light-blue;
+    margin-right: 10px;
+  }
+  &:hover{
+    color: $mm-visited-links;
```
Comment (besseym) [inline static/design_language/sass/design_language.scss]: Not an abstract color name? Looks like this color is being used for hover in addition to visited links.

File: static/design_language/sass/design_language.scss [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+  }
+}
+
+.sub-head{
+  //bootstrap
+  @extend .nav-link; //I hope I won't regret this decision
```
Comment (besseym) [inline static/design_language/sass/design_language.scss]: Yeah, comments about regret worry me.

File: static/design_language/sass/design_language.scss [hunk 7]
```diff
@@ -0,0 +1,191 @@
...
+    margin-left: 25px;
+    background-color: transparent;
+
+    &:hover{
+      color: $mm-visited-links;
+    }
+  }
+
+  //spacing
+  .collapsing > a.sub-head:last-child, .show > a:last-child{
```
Comment (besseym) [inline static/design_language/sass/design_language.scss]: You can nest these blocks as well using &

File: static/design_language/sass/design_language.scss [hunk 8]
```diff
@@ -0,0 +1,191 @@
...
+
+.left-nav{
+  //border-right: black solid;
+  //height: 100%;
+  box-shadow: 0.6rem 0rem 1rem 0 rgba(0,0,0,0.1);
+  overflow: hidden;
+}
+
+
+@media(min-width: 992px) { //makes them stack at small screen size
```
Comment (besseym) [inline static/design_language/sass/design_language.scss]: You should use Bootstrap's defined variable for small size.

File: webpack.config.js [hunk 9]
```diff
@@ -3,7 +3,8 @@ const webpack = require('webpack');
 module.exports = {
   entry: {
     app: __dirname + "/static/app/ts/index.ts",
-    app1: __dirname + "/static/app1/ts/index.ts"
+    app1: __dirname + "/static/app1/ts/index.ts",
+    designLanguage: __dirname + "/static/design_language/ts/index.ts"
```
Comment (besseym) [inline webpack.config.js]: signal

File: templates/design_language/base.html [hunk 10]
```diff
@@ -0,0 +1,146 @@
...
+                            <a class="sub-head" href="#">Principles</a>
+                            <a class="sub-head" href="#">Accessibility</a>
+                        </div>
+
+
+                        <a class="header collapsed mmui-label" id="style-content-header" data-toggle="collapse" href="#style-content" aria-expanded="false" aria-controls="style-content">
+                            <i class="fa fa-plus-circle"></i>Style & Layout
+                        </a>
+                        <div class="collapse" id="style-content">
+                            <a class="sub-head" href="#">Color</a>
```
Comment (besseym) [inline templates/design_language/base.html]: These should be in an h tag if they are intended as headers.

File: templates/design_language/base.html [hunk 11]
```diff
@@ -0,0 +1,146 @@
...
+        <!--closure for row-->
+    </div>
+
+
+
+
+    <script type="text/javascript" src="{{ url_for('static', filename='scripts/jquery/jquery-3.2.1.min.js') }}"></script>
+    <script type="text/javascript" src="{{ url_for('static', filename='scripts/popper/popper-1.12.9.min.js') }}"></script> <!-- Note this is from dist/umd, not dist. This is what bootstrap4 requires -->
+    <script type="text/javascript" src="{{ url_for('static', filename='scripts/bootstrap/bootstrap-4.0.0-beta.2.min.js') }}"></script>
+    <script type="text/javascript" src="{{ url_for('static', filename='scripts/designLanguage.js') }}"></script>
```
Comment (besseym) [inline templates/design_language/base.html]: signal

File: templates/design_language/base.html (hunk 10 above)
Comment (besseym) [reply templates/design_language/base.html]: But actually the look like nav items to me.

File: templates/design_language/base.html [hunk 12]
```diff
@@ -0,0 +1,146 @@
...
+                        </a>
+                        <div class="collapse" id="intro-content">
+                            <a class="sub-head {% if page == 'landing' %}active{% endif %}" href="/signal" >Goals</a>
+                            <a class="sub-head" href="#">Principles</a>
+                            <a class="sub-head" href="#">Accessibility</a>
+                        </div>
+
+
+                        <a class="header collapsed mmui-label" id="style-content-header" data-toggle="collapse" href="#style-content" aria-expanded="false" aria-controls="style-content">
+                            <i class="fa fa-plus-circle"></i>Style & Layout
```
Comment (besseym) [inline templates/design_language/base.html]: Use the character entity &amp;

File: templates/design_language/base.html [hunk 13]
```diff
@@ -0,0 +1,146 @@
...
+                    </a>
+
+
+                    <div class="mmui-input-group " style="width:66.66%; margin-bottom: 40px;">
+
+                        <input class=" mmui-form-control" placeholder="Search">
+
+                        <div class="input-group-btn" >
+                            <button class="btn " type="button">
+                                <i href="#" class="fa fa-search"></i>
```
Comment (besseym) [inline templates/design_language/base.html]: href?

File: templates/design_language/base.html [hunk 14]
```diff
@@ -0,0 +1,146 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <title>Siggy</title>
+
+    <link href="{{ url_for('static', filename='styles/design_language.css') }}" rel="stylesheet">
```
Comment (besseym) [inline templates/design_language/base.html]: signal

File: templates/design_language/base.html [hunk 15]
```diff
@@ -0,0 +1,146 @@
...
+                        </a>
+                        <div class="collapse" id="style-content">
+                            <a class="sub-head" href="#">Color</a>
+                            <a class="sub-head" href="#">Grid</a>
+                            <a class="sub-head" href="#">Icons</a>
+                            <a class="sub-head" href="#">Logo</a>
+                            <a class="sub-head" href="#">Responsive</a>
+                            <a class="sub-head" href="#">Typography</a>
+                            <a class="sub-head" href="#">Units and Measures</a>
+                            <a class="sub-head" href="#">Writing</a>
```
Comment (besseym) [inline templates/design_language/base.html]: Why don't these links have urls?

File: static/design_language/sass/design_language.scss [hunk 16]
```diff
@@ -0,0 +1,191 @@
...
+  padding-right: 1.5rem; //bad fix for scrollbar positioning. needs better config later.
+}
+
+.left-nav-row{
+  @extend .row;
+  height: 100%;
+}
+
+.left-nav::-webkit-scrollbar {
+display: none;
```
Comment (besseym) [inline static/design_language/sass/design_language.scss]: indent

File: static/design_language/sass/design_language.scss (hunk 2 above)
Comment (boyoonc) [reply static/design_language/sass/design_language.scss]: If I change .header to be .top-nav and .sub-head to be .sub-nav, I think this will be not weird!

**Comment by boyoonc** [comment]
would you review again please?

File: templates/app1/base.html [hunk 17]
```diff
@@ -88,7 +89,7 @@
         </div>
     </footer>
 
-    <!--this type of footer is designed to be taller which involves changes in mmui-theme, and the text style design is not finished.-->
+    <!--this type of footer is designed to be taller which involves changes in mmui-theme, and the text style design_language is not finished.-->
```
Comment (besseym) [inline templates/app1/base.html]: signal

File: templates/signal/base.html [hunk 18]
```diff
@@ -0,0 +1,146 @@
...
+    <div class="app-left-nav-container-fluid">
+        <!--not mmui-main-content as no footer in use-->
+        <div class="app-left-nav-row" >
+            <div class="col-md-3 app-left-nav" id="app-left">
+                <!--<div class="mmui-container-fluid">-->
+
+                    <br />
+
+                    <a href="/signal" class="app-left-nav-brand">
+                    <div style="margin-bottom: 80px">
```
Comment (besseym) [inline templates/signal/base.html]: class is better than inline styles

**Pull request merged**